odxtools/diaglayer.py

```diff
@@ -97,132 +64,79 @@ def __init__(
-        doc_frags.append(OdxDocFragment(short_name, "LAYER"))
-
-        odx_id = OdxLinkId.from_et(et_element, doc_frags)
-
-        # Parse DiagServices
-        services = [
-            DiagService.from_et(service, doc_frags)
-            for service in et_element.iterfind("DIAG-COMMS/DIAG-SERVICE")
-        ]
-        diag_comm_refs = []
-        for service in et_element.iterfind("DIAG-COMMS/DIAG-COMM-REF"):
-            ref = OdxLinkRef.from_et(service, doc_frags)
-            assert ref is not None
-            diag_comm_refs.append(ref)
-
-        single_ecu_jobs = [
-            SingleEcuJob.from_et(sej, doc_frags)
-            for sej in et_element.iterfind("DIAG-COMMS/SINGLE-ECU-JOB")
-        ]
-
-        # Parse ParentRefs
-        parent_refs = [
-            ParentRef.from_et(pr_el, doc_frags)
-            for pr_el in et_element.iterfind("PARENT-REFS/PARENT-REF")
-        ]
-
-        # Parse communication parameter refs
-        com_params = [
-            CommunicationParameterRef.from_et(el, doc_frags, variant_type)
-            for el in et_element.iterfind("COMPARAM-REFS/COMPARAM-REF")
-        ]
-
-        # Parse Requests and Responses
-        requests = []
-        for rq_elem in et_element.iterfind("REQUESTS/REQUEST"):
-            rq = create_any_structure_from_et(rq_elem, doc_frags)
-            assert isinstance(rq, Request)
-            requests.append(rq)
-
-        positive_responses = []
-        for pr_elem in et_element.iterfind("POS-RESPONSES/POS-RESPONSE"):
-            pr = create_any_structure_from_et(pr_elem, doc_frags)
-            assert isinstance(pr, Response)
-            positive_responses.append(pr)
-
-        negative_responses = []
-        for nr_elem in et_element.iterfind("NEG-RESPONSES/NEG-RESPONSE"):
-            nr = create_any_structure_from_et(nr_elem, doc_frags)
-            assert isinstance(nr, Response)
-            negative_responses.append(nr)
-
-        additional_audiences = [
-            AdditionalAudience.from_et(el, doc_frags)
-            for el in et_element.iterfind("ADDITIONAL-AUDIENCES/ADDITIONAL-AUDIENCE")
-        ]
-
-        functional_classes = [
-            FunctionalClass.from_et(el, doc_frags)
-            for el in et_element.iterfind("FUNCT-CLASSS/FUNCT-CLASS")
-        ]
-
-        state_charts = [
-            StateChart.from_et(el, doc_frags)
-            for el in et_element.iterfind("STATE-CHARTS/STATE-CHART")
-        ]
-
-        if et_element.find("DIAG-DATA-DICTIONARY-SPEC"):
-            diag_data_dictionary_spec = DiagDataDictionarySpec.from_et(
-                et_element.find("DIAG-DATA-DICTIONARY-SPEC"), doc_frags)
-        else:
-            diag_data_dictionary_spec = None
+        # Create DiagLayer
+        return DiagLayer(diag_layer_raw=diag_layer_raw)
+
+    #####
+    # <properties forwarded to the "raw" diag layer>
+    #####
+    @property
+    def variant_type(self) -> DiagLayerType:
+        return self.diag_layer_raw.variant_type
```

Comment:
Maybe use class inheritance instead of those property wrappers ?

Reply:
I'm not so sure if this is a good idea. do it in a later PR? Once this saga is finished, this should be pretty easy,,,